File: src/main/java/no/jstien/bikeapi/tsdb/read/TimeSerieJsonDeserializer.java
package no.jstien.bikeapi.tsdb.read;

import com.google.gson.*;

import java.lang.reflect.Type;

public class TimeSerieJsonDeserializer implements JsonDeserializer<TimeSerie> {
    @Override
    public TimeSerie deserialize(JsonElement jsonElement, Type type, JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
        JsonObject jsonObject = jsonElement.getAsJsonObject();

        TimeSerie timeSerie = new TimeSerie();

        jsonObject.keySet().forEach(key -> {
            long timestamp = Long.valueOf(key);
            long value = jsonObject.get(key).getAsLong();
            DataPoint dp = new DataPoint(timestamp, value);
            timeSerie.addDataPoint(dp);
        });

        return timeSerie;
    }
}

File: src/main/java/no/jstien/bikeapi/tsdb/read/TimeSerieJsonDeserializer.kt
package no.jstien.bikeapi.tsdb.read

import com.google.gson.*

import java.lang.reflect.Type

class TimeSerieJsonDeserializer : JsonDeserializer<TimeSerie> {
    @Throws(JsonParseException::class)
    override fun deserialize(jsonElement: JsonElement, type: Type, jsonDeserializationContext: JsonDeserializationContext): TimeSerie {
        val jsonObject = jsonElement.asJsonObject

        val timeSerie = TimeSerie()

        jsonObject.keySet().forEach { key ->
            val timestamp = java.lang.Long.valueOf(key)
            val value = jsonObject.get(key).asDouble
            val dp = DataPoint(timestamp, value)
            timeSerie.addDataPoint(dp)
        }

        return timeSerie
    }
}

